querydsl-jpa/src/main/java/com/querydsl/jpa/HibernateHandler.java

```diff
@@ -52,15 +53,15 @@ public boolean createNativeQueryTyped() {
     @SuppressWarnings("unchecked")
     @Override
     public <T> CloseableIterator<T> iterate(Query query, FactoryExpression<?> projection) {
-        if (query instanceof NativeQuery) {
-            NativeQuery hQuery = (NativeQuery) query;
-            ScrollableResults results = hQuery.scroll(ScrollMode.FORWARD_ONLY);
+        try {
+            org.hibernate.query.Query unwrappedQuery = query.unwrap(org.hibernate.query.Query.class);
```

Comment:
Remove instance of checks,  just attempt to unwrap. This is because the query may be wrapped in various layers of proxies.

Also, we should unwrap a `Query` here and not a `NativeQuery` because this code is also used for regular queries.